pkg/kube/controllers/extendedstatefulset/reconciler.go

```diff
@@ -211,6 +232,193 @@ func (r *ReconcileExtendedStatefulSet) Reconcile(request reconcile.Request) (rec
+		}
+
+		// Create the StatefulSet
+		if err := r.client.Create(ctx, &desiredOperatorStatefulSet); err != nil {
+			return errors.Wrapf(err, "could not create operator StatefulSet '%s' for ExtendedStatefulSet '%s' in namespace '%s'", desiredOperatorStatefulSet.Name, exStatefulSet.Name, exStatefulSet.Namespace)
+		}
+
+		ctxlog.Info(ctx, "Created Operator StatefulSet '", desiredOperatorStatefulSet.Name, "' for ExtendedStatefulSet '", exStatefulSet.Name, "' in namespace '", exStatefulSet.Namespace, "'.")
+
+		exStatefulSet.Spec.Template = *originalTemplate
+	}
+	return nil
+}
+
+// generateOperatorSingleStatefulSet creates a operator single statefulset per zone
+func (r *ReconcileExtendedStatefulSet) generateOperatorSingleStatefulSet(exStatefulSet *essv1a1.ExtendedStatefulSet, template *v1beta2.StatefulSet, zoneIndex int, zoneName string) (*v1beta2.StatefulSet, error) {
+
+	statefulSet := template.DeepCopy()
+
+	// Get the labels and annotations
+	labels := statefulSet.GetLabels()
+	if labels == nil {
+		labels = make(map[string]string)
+	}
+
+	annotations := statefulSet.GetAnnotations()
+	if annotations == nil {
+		annotations = make(map[string]string)
+	}
+
+	statefulSetNamePrefix := exStatefulSet.GetName()
+
+	// Update available-zone specified properties
+	if zoneIndex >= 0 && len(zoneName) != 0 {
+		// Reset name prefix with zoneIndex
+		statefulSetNamePrefix = fmt.Sprintf("%s-z%d", exStatefulSet.GetName(), zoneIndex)
+
+		labels[essv1a1.LabelAZIndex] = strconv.Itoa(zoneIndex)
+		labels[essv1a1.LabelAZName] = zoneName
+
+		zonesBytes, err := json.Marshal(exStatefulSet.Spec.Zones)
+		if err != nil {
+			return &v1beta2.StatefulSet{}, errors.Wrapf(err, "Could not marshal zones: '%v'", exStatefulSet.Spec.Zones)
+		}
+		annotations[essv1a1.AnnotationZones] = string(zonesBytes)
+
+		// Get the pod labels and annotations
+		podLabels := statefulSet.Spec.Template.GetLabels()
+		if podLabels == nil {
+			podLabels = make(map[string]string)
+		}
+		podLabels[essv1a1.LabelAZIndex] = strconv.Itoa(zoneIndex)
+		podLabels[essv1a1.LabelAZName] = zoneName
+
+		podAnnotations := statefulSet.Spec.Template.GetAnnotations()
+		if podAnnotations == nil {
+			podAnnotations = make(map[string]string)
+		}
+		podAnnotations[essv1a1.AnnotationZones] = string(zonesBytes)
+
+		statefulSet.Spec.Template.SetLabels(podLabels)
+		statefulSet.Spec.Template.SetAnnotations(podAnnotations)
+
+		statefulSet = r.updateAffinity(statefulSet, exStatefulSet.Spec.ZoneNodeLabel, zoneIndex, zoneName)
+	}
+
+	// Set updated properties
+	statefulSet.SetName(fmt.Sprintf("%s-%s", "operator", statefulSetNamePrefix))
+	statefulSet.SetLabels(labels)
+	statefulSet.SetAnnotations(annotations)
+
+	return statefulSet, nil
+}
+
+func (r *ReconcileExtendedStatefulSet) deleteOperatorStatefulSet(ctx context.Context, extendedstatefulset *essv1a1.ExtendedStatefulSet) error {
```

Comment:
These functions delete/update/etc. the "volume management" sts, correct?
A better name would be `deleteVolumeManagementStatefulSet`

Reply:
Sure. Done.